x/ccv/provider/keeper/msg_server.go

```diff
@@ -156,7 +158,32 @@ func (k msgServer) SubmitConsumerDoubleVoting(goCtx context.Context, msg *types.
 		return nil, err
 	}
 
-	if err := k.Keeper.HandleConsumerDoubleVoting(ctx, evidence, msg.InfractionBlockHeader.Header.ChainID); err != nil {
+	// parse the validator set of the infraction block header in order
+	// to find the public key of the validator who double voted
+
+	// get validator set
+	valset, err := tmtypes.ValidatorSetFromProto(msg.InfractionBlockHeader.ValidatorSet)
+	if err != nil {
+		return nil, err
+	}
+
+	// look for the malicious validator in the validator set
+	_, validator := valset.GetByAddress(evidence.VoteA.ValidatorAddress)
+	if validator == nil {
+		return nil, errorsmod.Wrapf(
+			ccvtypes.ErrInvalidEvidence,
+			"misbehaving validator %s cannot be found in the infraction block header validator set",
+			evidence.VoteA.ValidatorAddress)
+	}
+
+	pubkey, err := cryptocodec.FromTmPubKeyInterface(validator.PubKey)
+	if err != nil {
+		return nil, err
+	}
+
+	// handle the double voting evidence using the chain ID of the infraction block header
+	// and the malicious validator's public key
+	if err := k.Keeper.HandleConsumerDoubleVoting(ctx, evidence, msg.InfractionBlockHeader.Header.ChainID, pubkey); err != nil {
 		return &types.MsgSubmitConsumerDoubleVotingResponse{}, err
```

Comment:
In other cases where `err != nil` we return `return nil, err`. Here we return ` &types.MsgSubmitConsumerDoubleVotingResponse{}, err`.
Is this expected?

Reply:
Nice catch. It's not a problem since a non-nil error will always be returned along with `&types.MsgSubmitConsumerDoubleVotingResponse{}`. 
Thanks for pointing this out tho! I will change it for consistency.